cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -331,9 +340,15 @@ public void initialize(MasterEnvironmentContext context)
 
     // Services are publish to K8s with a prefix
     String resourcePrefix = "cdap-" + instanceName + "-";
+    List<String> loadBalancerServicelist = new ArrayList<>();
+    if (!Strings.isNullOrEmpty(conf.get(LOAD_BALANCER_SERVICES))) {
+      loadBalancerServicelist = Arrays.asList(
+          conf.get(LOAD_BALANCER_SERVICES).split(","));
+    }
     discoveryService = new KubeDiscoveryService(cdapInstallNamespace, "cdap-" + instanceName + "-",
-        podLabels,
-        podInfo.getOwnerReferences(), apiClientFactory);
+        podLabels, podInfo.getOwnerReferences(), apiClientFactory,
+        loadBalancerServicelist,
+        parseLoadBalancerAnnotations(conf.get(LOAD_BALANCER_SERVICE_ANNOTATIONS)));
```

Comment:
what are these annotations needed for?

Reply:
The implementation of load balancer services is dependent on cloud providers such as GCP, AWS etc. To configure cloud provider specific behaviour we need to specify annotations specific to the cloud provider. For example ["networking.gke.io/load-balancer-type" annotation](https://cloud.google.com/kubernetes-engine/docs/concepts/service-load-balancer#load_balancer_types) can be used in GCP to specify the type of load balancer to use. Added a comment explaining the same above the field declaration on top. Added a comment in the code.